modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/BusyExecutor.java

```diff
@@ -35,52 +39,65 @@
     private final String name;
 
     /** Active flag (used to skip commands in inactive cluster.) */
-    private volatile boolean active;
+    private final AtomicBoolean active = new AtomicBoolean(false);
 
     /** Lock protection of started gathering during deactivation. */
-    private final GridBusyLock busyLock = new GridBusyLock();
+    private volatile GridBusyLock busyLock = new GridBusyLock();
 
     /** Executor pool. */
     private final IgniteThreadPoolExecutor pool;
 
+    /** Cancellable tasks. */
+    private final ConcurrentMap<CancellableTask, Object> cancellableTasks = new ConcurrentHashMap<>();
```

Comment:
ConcurrentMap -> ConcurrentHashSet or ConcurrentSkipListSet ?

Reply:
CancellableTask does not implement Comparable so we can't use ConcurrentSkipListSet.
There is no ConcurrentHashSet structure in ApacheIgnite now.
But there is GridConcurrentHashSet (actually it is just a wrapper around ConcurrentHashMap and, unfortunately, it creates extra objects, but I replace ConcurrentHashMap with it.